server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -883,6 +887,7 @@ private void processBulkIndexIngestRequest(
                 } else {
                     long ingestTookInMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - ingestStartTimeInNanos);
                     BulkRequest bulkRequest = bulkRequestModifier.getBulkRequest();
+                    boolean bulkRequestNeedsClosing = bulkRequest != original;
```

Comment:
I don't like this. I think we should make it so that the refcounting is the same whether we get a new request or not, e.g. by adding a `bulkRequest.mustIncRef()` call to the branch in `BulkRequestModifier#getBulkRequest` which returns the original request. That way this method always has acquire-a-ref semantics so the caller always needs to decref its response.